src/services/inlayHints.ts

```diff
@@ -421,6 +444,258 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
         });
     }
 
+    function typeToInlayHintParts(type: Type): InlayHintDisplayPart[] | string {
+        if (!shouldUseInteractiveInlayHints(preferences)) {
+            return printTypeInSingleLine(type);
+        }
+
+        const flags = NodeBuilderFlags.IgnoreErrors | TypeFormatFlags.AllowUniqueESSymbolType | TypeFormatFlags.UseAliasDefinedOutsideCurrentScope;
+        const typeNode = checker.typeToTypeNode(type, /*enclosingDeclaration*/ undefined, flags);
+        Debug.assertIsDefined(typeNode, "should always get typenode");
+
+        const parts: InlayHintDisplayPart[] = [];
+        visitDisplayPart(typeNode);
+        function visitDisplayPart(node: Node) {

```

Comment:
nit: you're not visiting display parts, you're visiting nodes to aggregate display parts

Reply:
Yeah I just didn't want to call it `visitor` to differentiate it from the one at the top of `provideInlayHints`. Suggestions of a better name?

Reply:
`visitForDisplayParts`?